File: final_project-master/src/main/java/Openconnection/example/demo/beans/Company.java
package Openconnection.example.demo.beans;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import jakarta.validation.constraints.Email;
import lombok.*;

import java.util.List;

@Entity
@Table(name = "companies")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder

public class Company {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(nullable = false, unique = true)
    private String name;

    @Email
    @Column(nullable = false, unique = true, length = 40)
    private String email;

    @Column(nullable = false, length = 40)
    private String password;

    @Singular
    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true, mappedBy = "companyId")
    @JsonIgnore
    private List<Coupon> coupons;

}

File: CouponsWebsite/src/Components/Models/Company.ts
import { Coupon } from './Customer';
// Company.ts
export class Company {
  public id: number;
  public name: string;
  public email: string;
  public password: string;
  public coupons:Coupon[];

  constructor(id: number, name: string, email: string, password: string,coupons:Coupon[]=[]) {
    this.id = id;
    this.name = name;
    this.email = email;
    this.password = password;
    this.coupons=coupons;
  }

  // Getters
  public getId(): number {
    return this.id;
  }

  public getName(): string {
    return this.name;
  }
  public getCoupons(): number {
    return this.id;
  }
 
  public getEmail(): string {
    return this.email;
  }

  public getPassword(): string {
    return this.password;
  }
}
